protos/encodings.proto

```diff
@@ -193,7 +193,7 @@ message SimpleStruct {}
 
 // An array encoding for binary fields
 message Binary {
-  ArrayEncoding indices = 1;
+  Buffer indices = 1;
```

Comment:
Is this a breaking change to the format? What does that mean for users?

Reply:
I don't think it's a breaking change, as such. It won't affect users since the encoding and decoding workflows are still the same for binary types and there is no change from the user point of view
However we may have to revise this again once we use bitpacking for the indices encoding

Reply:
Aren't these protobuf messages serialized in the data files? I'm worried they won't be able to read Lance V2 files written with older versions. This seems to go against the protobuf advice for backwards compatibility:

> Almost never change the type of a field; it’ll mess up deserialization, same as re-using a tag number. The [protobuf docs](https://protobuf.dev/programming-guides/proto2#updating) outline a small number of cases that are okay (for example, going between int32, uint32, int64 and bool). However, changing a field’s message type will break unless the new message is a superset of the old one.

https://protobuf.dev/programming-guides/dos-donts/#change-type

Reply:
I see...yes, in that case it's a breaking change. But as per the current structure, we do need to be able to access the buffer index in the page if we want to coalesce requests, which makes changing the protobuf necessary (otherwise the string scheduling ends up taking too long during random access) .
Is there any recommended alternative in such a situation?

Reply:
I would create a new field and deprecate the old field. Or maybe create a `BinaryV2` message and deprecate `Binary` or something like that. You can either have an error message if it has the old one, or handle both fields/messages in different code paths. cc @westonpace

Reply:
I agree with @wjones127 that, at this point, we want to start avoiding these breaking changes.

Also, this approach won't work.  We want indices to be bitpacked which means we need this to be `ArrayEncoding`.

Reply:
I have updated the workflow to use `ArrayEncoding` but still schedule the requests together